File: Fuyuki.Server/Mapping/Group.cs
using System.Linq;
using AutoMapper;
using Fuyuki.Data;
using Fuyuki.ViewModels;

namespace Fuyuki.Mapping
{
    public class GroupProfile : Profile
    {
        public GroupProfile()
        {
            CreateMap<Group, GroupModel>();

            CreateMap<Group, GroupSubsModel>()
                .ForMember(x => x.Subreddits, opts => opts.MapFrom(x => x.GroupSubreddits.Select(s => s.Subreddit)));

            CreateMap<GroupModel, Group>()
                .ForMember(x => x.Name, opts => opts.MapFrom(x => x.Name.Trim()));

            CreateMap<GroupRequest, Group>()
                .ForMember(x => x.Name, opts => opts.MapFrom(x => x.Name.Trim()));

            CreateMap<GroupDto, GroupMembershipModel>();
        }
    }
}

File: Fuyuki.Client/src/interfaces/Group.ts
import { Subreddit } from './Subreddit';

export interface Group {
  id: number;
  name: string;
}

export interface GroupWithSubreddits extends Group {
  subreddits: Subreddit[];
}

export interface GroupMembership extends Group {
  isMember: boolean;
}
